File: SeLeAs/Java/equalRow.java
import java.util.*;

class equalRow {
    public int equalPairs(int[][] grid) {
        int n = grid.length;
        int pairs = 0;
        
        // Map to store the frequency of each row as a list
        Map<List<Integer>, Integer> rows = new HashMap<>();
        
        // Count occurrences of each row
        for (int[] row : grid) {
            List<Integer> rowList = new ArrayList<>();
            for (int val : row) {
                rowList.add(val);
            }
            rows.put(rowList, rows.getOrDefault(rowList, 0) + 1);
        }
        
        // Check each column
        for (int col = 0; col < n; col++) {
            List<Integer> columnList = new ArrayList<>();
            for (int row = 0; row < n; row++) {
                columnList.add(grid[row][col]);
            }
            if (rows.containsKey(columnList)) {
                pairs += rows.get(columnList);
            }
        }
        
        return pairs;
    }
}

File: SeLeAs/LeetCode/LeetCode 75/equalRow.py
from collections import defaultdict
from typing import List

class Solution:
    def equalPairs(self, grid: List[list[int]]) -> int:
        n = len(grid)
        pairs = 0
        rows = defaultdict(int)
        
        # Count occurrences of each row
        for row in grid:
            rows[tuple(row)] += 1
        
        # Check each column
        for col in range(n):
            column = tuple(grid[r][col] for r in range(n))
            if column in rows:
                pairs += rows[column]
        
        return pairs  # Move the return statement outside the column loop
